ex/auto/writer/odev_build_form/build_form.py

```diff
@@ -124,111 +135,89 @@ def create_form(self, doc: XTextDocument) -> None:
-            doc=BuildForm.doc,
-            name="Options",
+            doc=DOC,
+            name=name,
             label="Options",
             comp_kind=Forms.CompenentKind.GroupBox,
-            x=103,
-            y=5,
-            width=56,
+            x=col2_x,
+            y=y,
+            width=col2_width,
             height=25,
         )
 
         # these three radio buttons have the same name ("Option"), and
         # so only one can be on at a time
-        props = Forms.add_control(
-            doc=BuildForm.doc,
-            name="Options",
-            label="No automatic generation",
-            comp_kind=Forms.CompenentKind.RadioButton,
-            x=106,
-            y=11,
-            width=50,
-            height=6,
-        )
+        comp_kind = Forms.CompenentKind.RadioButton
+        indent = 3
+        height = 6
+        x = col2_x + indent
+        width = col2_width - 2 * indent
+
+        label = "No automatic generation"
+        y += height
+        props = Forms.add_control(DOC, name, label, comp_kind, x, y, width, height)
         props.addPropertyChangeListener("State", self)
 
-        props = Forms.add_control(
-            doc=BuildForm.doc,
-            name="Options",
-            label="Before inserting a record",
-            comp_kind=Forms.CompenentKind.RadioButton,
-            x=106,
-            y=17,
-            width=50,
-            height=6,
-        )
+        label = "Before inserting a record"
+        y += height
+        props = Forms.add_control(DOC, name, label, comp_kind, x, y, width, height)
         props.addPropertyChangeListener("State", self)
 
-        props = Forms.add_control(
-            doc=BuildForm.doc,
-            name="Options",
-            label="When moving to a new record",
-            comp_kind=Forms.CompenentKind.RadioButton,
-            x=106,
-            y=23,
-            width=50,
-            height=6,
-        )
+        label = "When moving to a new record"
+        y += height
+        props = Forms.add_control(DOC, name, label, comp_kind, x, y, width, height)
         props.addPropertyChangeListener("State", self)
 
         # check boxes inside another group box
         # use the same property change listener
-        Forms.add_control(
-            doc=BuildForm.doc,
-            name="Misc",
-            label="Miscellaneous",
-            comp_kind=Forms.CompenentKind.GroupBox,
-            x=103,
-            y=35,
-            width=56,
-            height=25,
-        )
-
-        props = Forms.add_control(
-            doc=BuildForm.doc,
-            name="DefaultDate",
-            label='Default sales date to "today"',
-            comp_kind=Forms.CompenentKind.CheckBox,
-            x=106,
-            y=39,
-            width=60,
-            height=6,
-        )
+        y = 33
+        width=60
+
+        name="Misc"
+        label="Miscellaneous"
+        comp_kind=Forms.CompenentKind.GroupBox
+        x=col2_x
+        y=35
```

Comment:
Should be `y=y` but there is a bug somewhere here.

Reply:
I would need a bit more information.
you can add a `return` statement for debugging purposes to what controls are drawn up to the `return` statement. This can help with debugging. For instance you could put a `return` statement before line `183`.
Although it best to use IDE debugger whenever possible.

Because there are other issues here that need to get cleaned up there is not a lot of sense in debuging until cleanup is done.
If there are issues at that point I will pull it into a new branch and debug it.

Reply:
`y=35` note wrong spacing:

![image](https://user-images.githubusercontent.com/11288701/195721730-d6c083a8-cd4d-4271-baaa-72411cd41c63.png)

`y=33` note layout does not reflect value (ie bug):

![image](https://user-images.githubusercontent.com/11288701/195722097-74b1b93e-b339-4085-b76b-8b328d1795e8.png)